src/bsoncxx/include/bsoncxx/docs/v1.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2009-present MongoDB, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if !defined(BSONCXX_PRIVATE_DOXYGEN_PREPROCESSOR)
+#error "This file is for documentation purposes only. It should not be included."
+#endif // !defined(BSONCXX_PRIVATE_DOXYGEN_PREPROCESSOR)
+
+///
+/// @file
+/// For documentation purposes only.
+///
+/// @note This header is not includable!
+///
+
+///
+/// @dir bsoncxx/v1
```

Comment:
There is no `bsoncxx/` subdirectory under `v1/` (e.g. `bsoncxx/v1/bsoncxx/`) as there is no future plans to update or remove the existing include directory paths to `bsoncxx/v_noabi/bsoncxx/` to support `<bsoncxx/header.hpp>` (unstable ABI interfaces). The `v_noabi` headers will continue to provide root namespace redeclarations even after stable ABI interfaces are released and the ABI version number is made stable. Only users who need to maintain ABI stability requirements will need to include ABI-specific headers (e.g. following an API breaking change that updates redeclarations from `v1 -> v2` while preserving the ABI stability of `v1` symbols).